### PR TITLE
Update 18.18 snapshot to specify dependencies for `weeder`

### PR DESCRIPTION
**Pull request by mjgpy3**
I broke our CI installation of weeder with #19:
```
Error: While constructing the build
plan, the following exceptions were
encountered:

In the dependencies for weeder-2.2.0:
    dhall-1.39.0 from stack
                 configuration does not
                 match ^>=1.30.0 || ^>=1.31.0 || ^>=1.32.0 || ^>=1.33.0 || ^>=1.34.0 || ^>=1.35.0 || ^>=1.36.0 || ^>=1.37.0 || ^>=1.40.0 
                 (latest matching
                 version is 1.40.2)
    generic-lens-2.1.0.0 from stack
                         configuration
                         does not
                         match ^>=1.1.0.0 || ^>=1.2.0.0 || ^>=2.0.0.0 || ^>=2.2.0.0 
                         (latest
                         matching
                         version
                         is 2.2.0.0)
needed since weeder is a build target.

Some different approaches to resolving
this:

  * Set 'allow-newer: true'
    in /root/.stack/config.yaml to ignore all version constraints and build anyway.

  * Recommended action: try adding the
    following to your extra-deps
    in /root/megarepo/backend/stack.yaml:

- dhall-1.40.2@sha256:db1d5f81912de498eeb8cf4535aebd76c695c016a8c5710970b61e83f3d345cb,34786
- generic-lens-2.2.0.0@sha256:4008a39f464e377130346e46062e2ac1211f9d2e256bbb1857216e889c7196be,3867

Plan construction failed.
```

This adds the exact dependencies, which successfully resolves the dependency plan (in local testing).

### CODE REVIEW

**Comment by pbrisbin** [comment]
So, we're holding ourselves back on `weeder` (because latest targets GHC 9) and thus need to hold these back too?

It's a valid use-case, I guess, for [`lsd`](https://github.com/freckle/lsd) to consider. Right now the warning is assuming the `extra-dep` was added to get a newer version early.

**Comment by mjgpy3** [comment]
@pbrisbin yeah, that's the approach I was intending here.

I'm happy to take other approaches if they're preferred (e.g. `extra-deps`).

**Comment by mjgpy3** [comment]
@pbrisbin I was curious if that was the case! Will update.

File: lts/18/18/FR1.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  - oidc-client-0.6.0.0@sha256:2079dc5c9dfb5b3e2fa93098254ca16787c01a0cd3634b1d84afe84c9a6c4825,3368
+  - pwstore-fast-2.4.4@sha256:9b6a37510d8b9f37f409a8ab3babac9181afcaaa3fce8ba1c131a7ed3de30698,1351
+
+  # For brittany-0.13.1.2
+  - data-tree-print-0.1.0.2@sha256:d845e99f322df70e0c06d6743bf80336f5918d5423498528beb0593a2afc1703,1620
+
+  # For stylish-haskell
+  - optparse-applicative-0.15.1.0@sha256:29ff6146aabf54d46c4c8788e8d1eadaea27c94f6d360c690c5f6c93dac4b07e,4810
+
+  # For weeder
```
Comment (pbrisbin) [inline lts/18/18/FR1.yaml:46]: Maybe note here that it's 2.2.0 and we're holding back because of GHC 9?

**Review by pbrisbin — APPROVED** [review]
Thanks! Sorry this process is so annoying.

File: lts/18/18/FR1.yaml (hunk 1 above)
Comment (mjgpy3) [reply lts/18/18/FR1.yaml]: Done, good idea!

**Pull request merged**